api/v1beta1/foundationdbcluster_types_test.go

```diff
@@ -2329,15 +2329,28 @@ var _ = Describe("[api] FoundationDBCluster", func() {
 					input: "127.0.0.1:bad",
 					err:   fmt.Errorf("strconv.Atoi: parsing \"bad\": invalid syntax"),
 				}),
-			Entry("IPv6 with invalid address",
+			Entry("host name",
 				testCase{
-					input: "[::1:]:4500",
-					err:   fmt.Errorf("invalid address: [::1:]:4500"),
+					input: "operator-test-1-storage-1.operator-test-1.my-ns.svc.cluster.local:4500",
+					expectedAddr: ProcessAddress{
+						StringAddress: "operator-test-1-storage-1.operator-test-1.my-ns.svc.cluster.local",
+						Port:          4500,
+						Flags:         nil,
+					},
+					expectedStr: "operator-test-1-storage-1.operator-test-1.my-ns.svc.cluster.local:4500",
+					err:         nil,
 				}),
-			Entry("IPv4 with invalid address",
+			Entry("IP from host name",
 				testCase{
-					input: "127.0.0.A:4500",
-					err:   fmt.Errorf("invalid address: 127.0.0.A:4500"),
+					input: "127.0.0.1:4501(fromHostname)",
+					expectedAddr: ProcessAddress{
+						IPAddress:    net.ParseIP("127.0.0.1"),
+						Port:         4501,
+						Flags:        nil,
+						FromHostname: true,
+					},
+					expectedStr: "127.0.0.1:4501(fromHostname)",
```

Comment:
If the provided input is an IP address port pair, why do we add the `(fromHostname)`? I would expect to either get an error or that the `(fromHostname)` suffix is removed/absent.

Reply:
This is a reflection of how the addresses are represented in the JSON status. @RenxuanW may be able to shed more light on the details there.

Reply:
Okay, that makes sense 👍